esbuild.js

```diff
@@ -0,0 +1,127 @@
+    serviceWorker: `${src}/background/service-worker-init.ts`,
+    debugTools: `${src}/debug-tools/initializer/debug-tools-init.tsx`,
+};
+const devWebExtensionOutdir = path.join(__dirname, 'extension/devBundle');
+const devWebExtensionM3Outdir = path.join(__dirname, 'extension/devMv3Bundle');
+
+const prodWebExtensionOutDir = path.join(__dirname, 'extension/prodBundle');
+
+// Default behavior; builds dev extension.
+let entryFiles = webExtensionEntryFiles;
+let outdir = devWebExtensionOutdir;
+let isProd = false;
+let platform = 'browser';
+let external = [];
+let format = 'esm';
+let define;
+let target = ['chrome90', 'firefox90'];
+let plugins = [CreateStylePlugin()];
+
+switch (argsObj.env) {
+    // Note: currently causes errors when electron app is run.
+    case 'unified':
+        entryFiles = electronEntryFiles;
+        outdir = unifiedOutdir;
+        platform = 'node';
+        external = [];
+        format = 'cjs';
+        break;
+
+    case 'prod':
+        isProd = true;
+        outdir = prodWebExtensionOutDir;
+        delete entryFiles.detailsView;
+        break;
+
+    case 'devM3':
+        outdir = devWebExtensionM3Outdir;
+        define = {
+            global: 'globalThis',
+        };
+        break;
+
+    case 'report':
+        entryFiles = { report: `${src}/reports/package/reporter-factory.ts` };
+        outdir = path.join(__dirname, 'packages/report/bundle');
+
+        // esbuild doesn't have an easy way to ignore node_modules for monorepos,
+        // so this plugin is necessary (marks all node_modules as external).
+        // Thread: https://github.com/evanw/esbuild/issues/619
+        plugins.push(
+            NodeResolve.NodeResolvePlugin({
+                extensions: ['.ts', '.tsx', '.js'],
+                onResolved: resolved => {
+                    if (resolved.includes('node_modules')) {
+                        return {
+                            external: true,
+                        };
+                    }
+                    return resolved;
+                },
+            }),
+        );
+        break;
+}
+
+const config = {
+    plugins,
+    entryPoints: entryFiles,
+    format,
+    platform,
+    external,
+    outbase: src,
+    bundle: true,
+    minify: isProd,
+    sourcemap: !isProd,
+    target,
+    outdir: outdir,
+    outExtension: {
+        '.js': '.bundle.js',
+    },
+    watch: argv.includes('--watch'),
+    logLevel: 'info',
+    define,
+};
+
+esbuild
+    .build(config)
+    .then(() => {
+        if (!isProd) {
+            return;
+        }
+
+        // minification of the details-view bundle leads to a collision with the axe-core package,
+        // so we do not minify that bundle specifically.
+        config.entryPoints = {
+            detailsView: `${src}/DetailsView/details-view-initializer.ts`,
+        };
+        config.minify = false;
+        esbuild.build(config).catch(() => process.exit(1));
+    })
+    .catch(() => process.exit(1));
```

Comment:
This catch handler (and the equivalent one on L125) feels like it would throw away error information that should probably be emitted to the console. Maybe easier to just use `buildSync` instead?

Reply:
Good catch! Changing to console.error for the catch.

On `buildSync`: it doesn't work when using plugins sadly.